### PR TITLE
Punching Mirror usage tracking

### PR DESCRIPTION
**Pull request by Glitched94**
- Made punching mirror behave similar to the school of hard knocks when used

### CODE REVIEW

**Comment by jaadams5** [comment]
Any chance you could make a test?

File: src/data/dailylimits.txt [hunk 1]
```diff
@@ -365,3 +365,4 @@ Use	vitachoconutriment capsule	_vitachocCapsulesUsed	3
 Use	warbear breakfast machine	_warbearBreakfastMachineUsed
 Use	warbear gyrocopter	_warbearGyrocopterUsed
 Use	warbear soda machine	_warbearSodaMachineUsed
+Use	punching mirror	_punchingMirrorUsed
```
Comment (midgleyc) [inline src/data/dailylimits.txt:368]: can you put this in alphabetical ordering?

**Comment by Glitched94** [comment]
> Any chance you could make a test?

In all honesty, I don't know how
I'll take a loot at the existing data in the test folder and see if I can work it out though, but I'd appreciate any pointers :)

**Comment by midgleyc** [comment]
Generally you want to find a test that looks close and then modify it.

`test/net/sourceforge/kolmafia/request/UseItemRequestTest.java`'s `successfulMilkUsageSetsPreferences` looks like a good pick to me.

**Review by midgleyc — APPROVED** [review]
Thanks!

**Review by midgleyc — CHANGES_REQUESTED** [review]
okay, so spotless has failed, could you run `./gradlew :spotlessApply` and re-commit: I hope that should do it.

**Comment by Glitched94** [comment]
> okay, so spotless has failed, could you run `./gradlew :spotlessApply` and re-commit: I hope that should do it.

Sorry for the delay on that, had to downgrade my java version to get gradle to run

**Comment by Glitched94** [comment]
I promise I'm not a _complete_ idiot

**Comment by Glitched94** [comment]
Switched IDEs and confirmed the test works locally, needed to add `withHippyStoneBroken()` to the test, false was the correct result without that

**Review by midgleyc — APPROVED** [review]

**Pull request merged**